riscv/sim.cc

```diff
@@ -414,8 +416,19 @@ void sim_t::idle()
 
   if (debug || ctrlc_pressed)
     interactive();
-  else
+  else {
+    if (instruction_limit.has_value()) {
+      if (*instruction_limit < INTERLEAVE) {
+        // Final step.
+        step(*instruction_limit);
+        exit(0);
+        *instruction_limit = 0;
+        return;
```

Comment:
I don't follow the logic here.  `exit(0)` shouldn't return, so can we delete the last two statements?

Reply:
Err yeah that is confusing, but it's actually calling `htif_t::exit()`, not `std::exit()`. There are a couple of calls in the constructor that do the same.

Probably should rename the method to avoid confusion, or we could do `this->exit(0);` but I kind of hate that.

Reply:
Ohh, I see.  Yeah, `this->exit(0)` would make this a lot clearer.

Reply:
Wouldn't `htif::exit(0)` work here?

Reply:
Or that.

Reply:
Changed to `htif_t::exit(0)`. I didn't touch the ones in the constructor - that code looks a bit wrong actually, probably due to exactly this confusion.

Reply:
Sorry, it's been so long since I wrote this... double checking it was actually me that added the `exit()` method, which is obviously a terrible idea (e.g. since it breaks the calls in the constructor! fun C++ footgun). I've renamed it to `htif_exit()`.

Reply:
SGTM.